dashboard/src/shared/AxiosInstance.ts

```diff
@@ -66,6 +66,15 @@ export function addErrorHandling(axiosInstance: AxiosInstance, store: Store<ISto
           // A 403 directly from the auth proxy requires reauthentication.
           if (Auth.usingOIDCToken() && Auth.is403FromAuthProxy(response)) {
             dispatchErrorAndLogout(message);
+            return Promise.reject(new UnauthorizedError(message));
+          }
+          // If not using auth proxy, an anonymous response is sent when
+          // a serviceaccount token expires (eg., delete secret xxx)
+          // or the token is managed externally (eg., vsphere kubectl login)
+          // In this case, force reauthentication
+          if (!Auth.usingOIDCToken() && Auth.isAnonymous(response)) {
+            dispatchErrorAndLogout(message);
+            return Promise.reject(new UnauthorizedError(message));
```

Comment:
these two ifs can be handled as one `if is403FromAuthProxy || isAnonymous` (the other comparison is irrelevant)

Also, why are you returning early? In theory `message` can be a JSON message (not sure for this use case) and it's handled later.

Reply:
Yes, I simply added another in case we should take other actions rather than logout. But, yes, if we gonna do the same in both alternatives we can merge them.

Regarding the rejection, I just thought that the logout action would be the last action and, consequently, there's no need to try to parse the JSON message (particularly, when it is an anonymous 403 it will always fail). But ok, I'll remove the early rejections just in case.